dev/io.openliberty.mcp.internal/src/io/openliberty/mcp/internal/ToolMetadata.java

```diff
@@ -58,11 +54,11 @@ private static Map<String, ArgumentMetadata> getArgumentMap(AnnotatedMethod<?> m
         for (AnnotatedParameter<?> p : method.getParameters()) {
             ToolArg pInfo = p.getAnnotation(ToolArg.class);
             if (pInfo != null) {
-                ArgumentMetadata pData = new ArgumentMetadata(p.getBaseType(), p.getPosition(), pInfo.description(), pInfo.required());
-                if (pInfo.name().equals(Tool.ELEMENT_NAME)) {
-                    result.put(method.getJavaMember().getName(), pData);
-                } else {
-                    result.put(pInfo.name(), pData);
+                String toolArgName = (pInfo.name().equals(Tool.ELEMENT_NAME)) ? p.getJavaParameter().getName() : pInfo.name(); // p.getJavaParameter().getName() needs java compiler -parameter flag to work
+                ArgumentMetadata ArgInfo = new ArgumentMetadata(p.getBaseType(), p.getPosition(), pInfo.description(), pInfo.required(), false);
+                boolean isDuplicateArg = result.put(toolArgName, ArgInfo) != null;
+                if (!toolArgName.isBlank() && isDuplicateArg) {
+                    result.put(toolArgName, new ArgumentMetadata(p.getBaseType(), p.getPosition(), pInfo.description(), pInfo.required(), isDuplicateArg));
```

Comment:
```suggestion
                boolean isDuplicateArg = result.contains(toolArgName);
                ArgumentMetadata argInfo = new ArgumentMetadata(p.getBaseType(), p.getPosition(), pInfo.description(), pInfo.required(), isDuplicateArg);
                result.put(toolArgName, argInfo);
```

It's more straightforward to just check if the argument name is a duplicate before adding it to the map, rather than adding it, and then updating it if it was a duplicate.

`ArgInfo` -> `argInfo` because it's a variable name.

I'm not sure what the purpose of the if `!toolArgName.isBlank()` check was.

Reply:
Aah, yes. Good catch. I could have just done a get as well. 
I'll use boolean:
`isDuplicateArg = result.containsKey(toolArgName); `
as contains() doesn't exist.

Reply:
Ah yes, `containsKey` is what you want :+1: